website/source/docs/guides/creating-certificates.html.md

```diff
@@ -32,140 +32,141 @@ signed by this CA will be allowed to communicate with the cluster.
+In order to authenticate Consul servers, server certificates are provided with a
+special certificate - one that contains `server.dc1.consul` in the `Subject
+Alternative Name`. If you enable `verify_server_hostname`, only agents that can
+provide such certificate are allowed to boot as a server. An attacker could
+otherwise compromise a Consul agent and restart the agent as a server in order
+to get access to all the data in your cluster! This is why server certificates
+are special, and only servers should have them provisioned.
 
-* `client.global.consul` for a client node in the `global` region
-* `server.us-west.consul` for a server node in the `us-west` region
+### Step 2: Create Client Certificates
 
-To create certificates for the client and server in the cluster with
-[cfssl][cfssl], create the following configuration file as `cfssl.json` to increase the default certificate expiration time:
+Create a client certificate:
 
-```json
-{
-  "signing": {
-    "default": {
-      "expiry": "87600h",
-      "usages": [
-        "signing",
-        "key encipherment",
-        "server auth",
-        "client auth"
-      ]
-    }
-  }
-}
+```shell
+$ consul tls cert create -client
+==> Using consul-ca.pem and consul-ca-key.pem
+==> Saved consul-client-0.pem
+==> Saved consul-client-0-key.pem
 ```
 
-```shell
-# Generate a certificate for the Consul server
-$ echo '{"key":{"algo":"rsa","size":2048}}' | cfssl gencert -ca=consul-ca.pem -ca-key=consul-ca-key.pem -config=cfssl.json \
-    -hostname="server.global.consul,localhost,127.0.0.1" - | cfssljson -bare server
+Client certificates are also signed by your CA, but they do not have that
+special `Subject Alternative Name` which means that if `verify_server_hostname`
+is enabled, they cannot start as a server.
+
+### Step 3: Create CLI Certificates [Optional]
 
-# Generate a certificate for the Consul client
-$ echo '{"key":{"algo":"rsa","size":2048}}' | cfssl gencert -ca=consul-ca.pem -ca-key=consul-ca-key.pem -config=cfssl.json \
-    -hostname="client.global.consul,localhost,127.0.0.1" - | cfssljson -bare client
+If you enforce HTTPS you will need a certificate in order to use consul commands
+or curl to access the HTTPS API.
 
-# Generate a certificate for the CLI
-$ echo '{"key":{"algo":"rsa","size":2048}}' | cfssl gencert -ca=consul-ca.pem -ca-key=consul-ca-key.pem -profile=client \
-    - | cfssljson -bare cli
+Create a CLI certificate:
+
+```shell
+$ consul tls cert create -cli
+==> Using consul-ca.pem and consul-ca-key.pem
+==> Saved consul-cli-0.pem
+==> Saved consul-cli-0-key.pem
 ```
 
+### Note on SANs for Server and Client Certificates
+
 Using `localhost` and `127.0.0.1` as subject alternate names (SANs) allows
 tools like `curl` to be able to communicate with Consul's HTTP API when run on
 the same host. Other SANs may be added including a DNS resolvable hostname to
 allow remote HTTP requests from third party tools.
 
+### What goes where
```

Comment:
Maybe, "Step 4: What goes where

I generally really like this section, but I think it might be nice to be a bit more specific about the file locations.

Reply:
This is part of the rewritten sections. I tried to be as specific as possible, but this is highly dependent on the existing setups.